src/main/java/ch/njol/skript/SkriptEventHandler.java

```diff
@@ -251,35 +249,40 @@ public static void registerBukkitEvent(Trigger trigger, Class<? extends Event> e
 	 * @param trigger The Trigger to unregister events for.
 	 */
 	public static void unregisterBukkitEvents(Trigger trigger) {
-		triggers.removeIf(pair -> {
-			if (pair.getSecond() != trigger)
-				return false;
+		Iterator<Map.Entry<Class<? extends Event>, Trigger>> entryIterator = triggers.entries().iterator();
+		Map.Entry<Class<? extends Event>, Trigger> entry;
+		entryLoop: while (entryIterator.hasNext()) {
+			entry = entryIterator.next();
+			if (entry.getValue() != trigger)
+				continue;
+			Class<? extends Event> event = entry.getKey();
 
-			HandlerList handlerList = getHandlerList(pair.getFirst());
-			assert handlerList != null;
+			// Remove the trigger from the map
+			entryIterator.remove();
 
+			// check if we can unregister the listener
 			EventPriority priority = trigger.getEvent().getEventPriority();
-			if (triggers.stream().noneMatch(pair2 ->
-				trigger != pair2.getSecond() // Don't match the trigger we are unregistering
-				&& pair2.getFirst().isAssignableFrom(pair.getFirst()) // Basic similarity check
-				&& priority == pair2.getSecond().getEvent().getEventPriority() // Ensure same priority
-				&& handlerList == getHandlerList(pair2.getFirst()) // Ensure same handler list
-			)) { // We can attempt to unregister this listener
-				Skript skript = Skript.getInstance();
-				for (RegisteredListener registeredListener : handlerList.getRegisteredListeners()) {
-					Listener listener = registeredListener.getListener();
-					if (
-						registeredListener.getPlugin() == skript
-						&& listener instanceof PriorityListener
-						&& ((PriorityListener) listener).priority == priority
-					) {
-						handlerList.unregister(listener);
-					}
+			for (Trigger t : triggers.get(event)) {
+				if (t.getEvent().getEventPriority() == priority) {
+					continue entryLoop;
 				}
 			}
 
-			return true;
-		});
+			// We can attempt to unregister this listener
+			HandlerList handlerList = getHandlerList(event);
+			assert handlerList != null;
+			Skript skript = Skript.getInstance();
+			for (RegisteredListener registeredListener : handlerList.getRegisteredListeners()) {
+				Listener listener = registeredListener.getListener();
+				if (
+					registeredListener.getPlugin() == skript
+					&& listener instanceof PriorityListener
+					&& ((PriorityListener) listener).priority == priority
+				) {
+					handlerList.unregister(listener);
+				}
```

Comment:
Should be 3 continues instead of multiple conditions all at once.

Reply:
What's wrong with the three at once?

Reply:
one condition at a time is faster, and it will be way better to read

Reply:
I get the readability concern, but how would it be faster? Java short-circuits as soon as it evaluates a `false`, so there shouldn't be any speed differences as far as I'm aware.

Reply:
You continue onto the next condition whereas a continue would end and continue, your case may be fine as most of the time the first condition should stop the trigger. Continue still required.

Reply:
Short-circuiting means the even in the current state, the second condition will not be evaluated if the first returns false.
So I don't really see why continue would make that any faster, but I'll change it for the readability, I guess.

Reply:
if true == true && true == true 3 operations

if true == true
    continue

2 operations

Reply:
&& is a fast-fail operation, if the first part of the condition is false it stops evaluating and moves on because there is no way it can be true.

The condition from this code
```java
var var = true;
if (var && var) // do the thing
```
is compiled as follows
```cpp
    iload 1
    ifeq l0 // if var is false, go to l0
    iload 1
    ifeq l0 // if var is false, go to l0
    // do the thing
   l0
    // move on
```

As you can see the conditions are checked one at a time on their failure condition (being false) so if the first part of the condition fails the second part will not be evaluated.

This means that code like
```java
if (false && methodThatThrowsException()) // do something
```
won't throw an error since the condition will fail at `false` and jump past the second part.

Reply:
Actually now that I write it out, I think it's less readable in the `continue` form as well:
```
                        for (RegisteredListener registeredListener : handlerList.getRegisteredListeners()) {
				Listener listener = registeredListener.getListener();
				if (registeredListener.getPlugin() != skript)
					continue;

				if (!(listener instanceof PriorityListener))
					continue;

				if (((PriorityListener) listener).priority != priority)
					continue;

				handlerList.unregister(listener);
			}
```
I'd like another opinion or two, but honestly I think it's better the way it is now.

Reply:
IMO the way it's right now it good

Reply:
Naw, that's totally way better

Reply:
It's good as it is now. There isn't a performance nor a readability concern to me.

Reply:
There is